### PR TITLE
#167938539 terms conditions frontend 

### PR DESCRIPTION
**Pull request by murediane**
#### What does this PR do?
implement terms and conditions page
#### Description of Task to be completed?
The terms conditions redux
Terms and conditions  react page 
unit test
#### How should this be manually tested?
 run the app and open /termsandconditions
#### Any background context you want to provide?
A user needs to know the terms and conditions before agreeing on signup ..this  feature is to make the signup  fully complete 
#### What are the relevant pivotal tracker stories?
[167938539](https://www.pivotaltracker.com/story/show/167938539)
#### Screenshots (if appropriate)
Screen Shot 2019-08-22 at 15 52 26

### CODE REVIEW

File: src/App/Components/TermsAndConditions/index.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      </div>
+    );
+  }
+}
+TermsAndConditions.propTypes = {
+  getTerms: propTypes.func.isRequired,
+  termsAndCondReducer: propTypes.object
+};
+export const mapStateToProps = ({ termsAndCondReducer }) => ({ termsAndCondReducer });
+export const mapDispatchToProps = dispatch => ({ getTerms: () => dispatch(terms()) });
```
Comment (bdushimi) [inline src/App/Components/TermsAndConditions/index.js:35]: @murediane looks a little bit complicated though it works. I'd suggest using bindActionCreators inside mapDispatchToProps for more clean & readable code.

File: src/App/_tests_/Redux/Reducers/termsreducer.test.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import termsReducer from '../../../../Redux/Reducers/termsAndcondReducers';
+import actions from '../../../../Redux/Actions';
+
+describe('TERMS Reducer', () => {
+  it('FETCH_TERMS_SUCCESS reducer', () => {
+    const initialState = { termsAndConditions: '' };
+    expect(termsReducer(initialState, {
+      type: actions.FETCH_TERMS_SUCCESS,
+      payload: { termsAndConditions: '' }
```
Comment (bdushimi) [inline src/App/_tests_/Redux/Reducers/termsreducer.test.js]: @murediane this test is not updating the state, therefore the reducer is not properly tested.

File: src/Redux/Reducers/index.js [hunk 3]
```diff
@@ -3,10 +3,12 @@ import testReduxReducer from './testReduxReducer';
...
 import authReducer from './auth';
+import termsAndCondReducer from './termsAndcondReducers';
 
 export default combineReducers({
   authReducer,
   testRedux: testReduxReducer,
   passwordReset,
-  registration
+  registration,
+  termsAndCondReducer
```
Comment (bdushimi) [inline src/Redux/Reducers/index.js]: @murediane meaningful variable naming! It would be more readable to name it as termsAndConditionReducer

**Review by bdushimi — CHANGES_REQUESTED** [review]

File: src/App/Components/TermsAndConditions/index.js (hunk 1 above)
Comment (murediane) [reply src/App/Components/TermsAndConditions/index.js:35]: I was comfortable with this, and I am willing to make it clear to whoever is not getting it since I am not familiar with the bindActionCreators.

File: src/App/_tests_/Redux/Reducers/termsreducer.test.js (hunk 2 above)
Comment (murediane) [reply src/App/_tests_/Redux/Reducers/termsreducer.test.js]: Can you please explain exactly what you mean by it is not updating the state

Comment (murediane) [reply src/App/_tests_/Redux/Reducers/termsreducer.test.js]: I have changed the test

File: src/App/Components/TermsAndConditions/index.js (hunk 1 above)
Comment (murediane) [reply src/App/Components/TermsAndConditions/index.js:35]: and Based to the research I made about the bindActionCreators I don't think I need them since I am using  Redux
https://redux.js.org/api/bindactioncreators#bindactioncreatorsactioncreators-dispatch

File: src/App/App.js [hunk 4]
```diff
@@ -47,6 +48,7 @@ class App extends Component {
               component={!localStorage.getItem('token') ? ResetPassword : Home}
             />
             <Route exact path="/redirect" component={Redirection} />
+            <Route exact path="/termsandconditions" component={TermsAndconditions} />
```
Comment (nkpremices) [inline src/App/App.js]: please @murediane 
Use snake case camelCase snake_case

File: src/App/Components/Registration/Registration.js [hunk 5]
```diff
@@ -158,7 +158,7 @@ export class Registration extends Component {
               />
               Agree to
               {' '}
-              <a href="/terms">terms and conditions</a>
+              <a href="/termsandconditions">terms and conditions</a>
```
Comment (alainburindi) [inline src/App/Components/Registration/Registration.js]: can you please change this to Link?? it's a mistake from me

File: src/Style/Sass/component/_navProfile.scss [hunk 6]
```diff
@@ -12,7 +12,7 @@
   &__name {
     color: $primary-white;
     display: inline;
-    font-family: Georgia, 'Times New Roman', Times, serif;
+    font-family: Georgia, "Times New Roman", Times, serif;
```
Comment (nkpremices) [inline src/Style/Sass/component/_navProfile.scss:15]: Change this font to lato ? @murediane

**Review by nkpremices — APPROVED** [review]
Looks good @murediane 
But please check out what I suggested

File: src/Redux/Actions/terms.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    dispatch({ type: actions.FETCH_TERMS_START });
+    const { data } = await axios.get('/api/termsAndConditions/1');
+    const { termsDocument: { termsAndConditions } } = data;
+    dispatch({
+      type: actions.FETCH_TERMS_SUCCESS,
+      payload: { termsAndConditions }
+    });
+  } catch (error) {
+    dispatch({ type: actions.FETCH_TERMS_FAIL });
+  }
```
Comment (alainburindi) [inline src/Redux/Actions/terms.js]: you are dispatching `FETCH_TERMS_START` and `FETCH_TERMS_FAIL` but you are not listening to them in you reducer

File: src/Style/Sass/component/_navProfile.scss (hunk 6 above)
Comment (nkpremices) [reply src/Style/Sass/component/_navProfile.scss:15]: Never mind. This font is for the logo

File: src/Style/Sass/layout/_footer.scss [hunk 8]
```diff
@@ -15,6 +15,9 @@
     left: 40%;
   }
 }
+.custom-footer {
```
Comment (alainburindi) [inline src/Style/Sass/layout/_footer.scss]: you are not using this

**Review by bdushimi — APPROVED** [review]

File: src/Style/Sass/abstract/_variables.scss [hunk 9]
```diff
@@ -9,3 +9,4 @@ $form-button-color: #2f3640;
 $text-color-default: #000000;
 $primary-theme-color: #2f3640;
 $primary-white: white;
+$primary-theme-color: #2f3640;
```
Comment (alainburindi) [inline src/Style/Sass/abstract/_variables.scss]: why did you add this color twice?

**Review by alainburindi — CHANGES_REQUESTED** [review]

**Review by alainburindi — APPROVED** [review]
